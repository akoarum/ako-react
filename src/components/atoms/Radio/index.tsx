import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'

type Props = {
  className?: string
  name: string
  value: string
  checked?: boolean
  required?: boolean
  onChange?: (e: React.FormEvent<HTMLInputElement>) => void
}

export const StyledRadio = styled.label`
  position: relative;

  .radio__input {
    position: absolute;
    left: 0;
    top: 0;
    width: 5px;
    height: 5px;
    opacity: 0;

    &:checked {
      + span::before {
        border-color: ${style.color.theme};
      }
      + span::after {
        opacity: 1;
      }
    }
  }

  .radio__label {
    &::before,
    &::after {
      position: absolute;
      top: 50%;
      content: '';
      border-radius: 50%;
      transform: translate3d(0, -50%, 0);
    }

    &::before {
      left: 0;
      border: 1px solid ${style.color.border};
      transition: border-color 0.1s $easing;
    }

    &::after {
      background: ${style.color.theme};
      opacity: 0;
      transition: opacity 0.1s ${style.animation.easing};
    }
  }

  ${style.mediaQuery.pc} {
    padding: 7px 0 6px 32px;
    font-size: ${style.font.sizePc};

    .radio__label {
      &::before {
        width: 24px;
        height: 24px;
      }
      &::after {
        left: 6px;
        width: 14px;
        height: 14px;
      }
    }
  }

  ${style.mediaQuery.sp} {
    padding: 7px 0 6px 28px;
    font-size: ${style.font.sizeSp};

    .radio__label {
      &::before {
        width: 20px;
        height: 20px;
      }
      &::after {
        left: 5px;
        width: 12px;
        height: 12px;
      }
    }
  }
`

export const Radio: React.FC<Props> = props => {
  return (
    <StyledRadio className={props.className}>
      <input
        type="radio"
        name={props.name}
        value={props.value}
        required={props.required}
        checked={props.checked}
        className="radio__input"
        onChange={props.onChange}
      />
      <span className="radio__label">{props.children}</span>
    </StyledRadio>
  )
}

export default Radio
