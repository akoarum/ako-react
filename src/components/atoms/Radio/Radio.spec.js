import React from 'react'
import renderer from 'react-test-renderer'
import Radio from './index'

describe('Radio', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderer.create(
      <Radio name="name" value="value">テスト</Radio>
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
