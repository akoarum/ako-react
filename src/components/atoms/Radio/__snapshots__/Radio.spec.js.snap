// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio initialize snapshot 1`] = `
<label
  className="sc-bdVaJa isRIPN"
>
  <input
    className="radio__input"
    name="name"
    type="radio"
    value="value"
  />
  <span
    className="radio__label"
  >
    テスト
  </span>
</label>
`;
